Source/Scene/ShadowVolumeAppearanceShader.js

```diff
@@ -0,0 +1,347 @@
+define([
+        '../Core/Check',
+        '../Core/defaultValue',
+        '../Core/defineProperties',
+        '../Renderer/PixelDatatype',
+        '../Scene/PerInstanceColorAppearance'
+], function(
+        Check,
+        defaultValue,
+        defineProperties,
+        PixelDatatype,
+        PerInstanceColorAppearance) {
+    'use strict';
+
+    var shaderDependenciesScratch = new ShaderDependencies();
+    /**
+     * Creates the shadow volume fragment shader for a ClassificationPrimitive to use a given appearance.
+     *
+     * @param {Appearance} appearance An Appearance to be used with a ClassificationPrimitive.
+     * @param {Boolean} [extentsCulling=false] Discard fragments outside the instance's spherical extents.
+     * @returns {String} Shader source for a fragment shader using the input appearance.
+     * @private
+     */
+    function ShadowVolumeAppearanceShader(appearance, extentsCulling, planarExtents) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.object('appearance', appearance);
+        //>>includeEnd('debug');
+
+        this._extentsCulling = defaultValue(extentsCulling, false);
+        this._planarExtents = defaultValue(planarExtents, false);
+        this._shaderSource = createShadowVolumeAppearanceShader(appearance, this._extentsCulling, this._planarExtents);
+        this._usesTexcoords = shaderDependenciesScratch._requiresTexcoords;
+    }
+
+    defineProperties(ShadowVolumeAppearanceShader.prototype, {
+        /**
+         * Whether or not the resulting shader uses texture coordinates.
+         *
+         * @memberof ShadowVolumeAppearanceShader.prototype
+         * @type {Boolean}
+         * @readonly
+         */
+        usesTexcoords : {
```

Comment:
Throughout perhaps `TextureCoordinates` to avoid the abbreviation.  That would definitely be the cases for a public API, but even for a private API, I don't think it is too verbose.